src/core/c-do.c

```diff
@@ -446,12 +446,12 @@ void Trace_Arg(REBINT num, const REBVAL *arg, const REBVAL *path)
 		Recycle();
 	}
 
-#ifdef NOT_USED_INVESTIGATE
-	if (GET_FLAG(sigs, SIG_EVENT_PORT)) {  // !!! Why not used?
+	// Was #ifdef'd out in R3-Alpha open source code. Enabled due to #1422
+	if (GET_FLAG(sigs, SIG_EVENT_PORT)) {
+		REBINT result = Awake_System(NULL, FALSE); // !!! just event port?
 		CLR_SIGNAL(SIG_EVENT_PORT);
-		Awake_Event_Port();
+		cast(void, result); // !!! Should the result be acted on?
 	}
-#endif
 
```

Comment:
This will cause unexpected recursions. Think about a button with an action function. When the button is clicked, the action function could call Do_Singals and then Awake_System, which handles the pending events. If there are some  pending clicks on this button, the same action function will then be called and cause recursive calls. This is similar to the recursion I wanted to avoid with wait/only.

Basically, we'd like the Awake_System is only be called from WAIT, so that the user knows there is only one place that event handling could happen.